industrial_ci/src/workspace.sh

```diff
@@ -190,8 +190,23 @@ function ici_import_url {
 function  ici_import_directory {
     local sourcespace=$1; shift
     local dir=$1; shift
-    rm -rf "$sourcespace:?/$(basename "$dir")"
-    cp -a "$dir" "$sourcespace"
+    local target
+    target=${sourcespace:?}/$(basename "$dir")
+
+    rm -rf "$target"
+    mkdir "$target"
+    local args=()
+    for p in "$BASEDIR" "$CCACHE_DIR" "$(readlink -m "$ICI_SRC_PATH/../..")"; do
+        if [ -n "$p" ]; then
+            args+=("(" "!" -path "$p" ")")
+        fi
+    done
+    while read -r f; do
+        local ft
+        ft=$target/$(basename "$f")
+        echo "$f -> $ft"
+        cp -a "$f" "$ft"
+    done < <(find "$dir" -maxdepth 1 -mindepth 1 "${args[@]}")
```

Comment:
I'll test this but I suspect this will not work if the user places the BASEDIR or CCACHE_DIR at any depth except right below the repo directory.  That is why I switched to the `tar` command over `find`.  For example if the user does something like this:

```
CCACHE_DIR: .cache/ccache
BASEDIR: .cache/workspaces
```

In my usecase I only wanted to cache the upstream workspace and not all the data in the basedir so placing everything in a `.cache` directory and caching that didn't work well for my use case.  However, this still revealed the issue with find and the inability to deal with directories that were deeper than just the first level.  Maybe there is a way to do this with find, I just don't know.  Find seems to have it's own language spec that is very confusing to me.

Reply:
> I'll test this but I suspect this will not work if the user places the BASEDIR or CCACHE_DIR at any depth except right below the repo directory. 

Ah, yes,  I think you are correct.

> That is why I switched to the tar command over find

It should be possible with `find` as well.
I will give it a try.

Reply:
> It should be possible with find as well.
> I will give it a try.

It is possible with path pruning and special handling of directories.
But tar is simpler and will perform better.
And since this is now run in the target container a suitable tar version should always ne pre-installed.

@tylerjw: Please review/try the updated version.

Reply:
Tested and this seems to work for me.  Thank you for following up on this!